### PR TITLE
add packet redundancy library

### PR DESCRIPTION
**Pull request by basvs**

### CODE REVIEW

File: redundancy/redundancy.c [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+			v ^= gbf_mul(*data++, x[j]);
+		*out++ = v;
+	}
+}
+
+void
+gbf_decode(gbf_int_t *out, gbf_int_t *data, gbf_int_t *vec, int num_frag, int size)
+{
+	// NOTE: this can grow huge..
+	gbf_int_t *x = (gbf_int_t *) alloca(sizeof(gbf_int_t) * num_frag * num_frag);
```
Comment (Spritetm) [inline redundancy/redundancy.c]: Can you either make a function that accepts a static buffer for this or use malloc() and friends? I don't want to give bpp a ginormous stack if I can help it...

File: redundancy/redundancy.c [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+				row[k] = row[k+1] ^ gbf_mul(mul, invrow[k]);
+			row[size-1] = gbf_mul(mul, invrow[k]);
+        }
+    }
+}
+
+void
+gbf_encode_one(gbf_int_t *out, gbf_int_t *data, gbf_int_t vec, int num_frag, int size)
+{
+	if (vec == 0)
```
Comment (Spritetm) [inline redundancy/redundancy.c]: Feel free to assert() for cases like this.

Comment (basvs) [reply redundancy/redundancy.c]: good idea; will fix that

File: redundancy/redundancy.c (hunk 1 above)
Comment (basvs) [reply redundancy/redundancy.c]: it now does alloca() for <= 256 bytes (is defined with a #define) and malloc()/free() for more memory.

File: redundancy/redundancy.c [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+		{
+			int k;
+			gbf_int_t v = 0;
+			for (k=0; k<num_frag; k++)
+				v ^= gbf_mul(data[k*size + j], x[i*num_frag + k]);
+			out[j*num_frag + i] = v;
+		}
+	}
+
+	if (sizeof(gbf_int_t) * num_frag > MAX_STACK_ALLOC) {
```
Comment (Spritetm) [inline redundancy/redundancy.c]: You're missing a '*num_frag' here.

File: redundancy/redundancy.h [hunk 4]
```diff
@@ -0,0 +1,73 @@
+#ifndef REDUNDANCY_H
+#define REDUNDANCY_H
+
+#include <stdint.h>
+
+#define GBF_BITS 16
```
Comment (Spritetm) [inline redundancy/redundancy.h]: Can you add  some explanation to what this does?

File: redundancy/redundancy.c (hunk 3 above)
Comment (basvs) [reply redundancy/redundancy.c]: aaargh.. thanks!

File: redundancy/redundancy.h (hunk 4 above)
Comment (basvs) [reply redundancy/redundancy.h]: tried to. it's just the size of the finite field.

**Pull request merged**